StreamChatClient_v3/WebSocketClient/Engine/StarscreamWebSocketEngine.swift

```diff
@@ -0,0 +1,64 @@
+//
+// StarscreamWebSocketEngine.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+#if canImport(Starscream)
+import Starscream
+
+final class StarscreamWebSocketProvider: WebSocketEngine {
+  private let webSocket: Starscream.WebSocket
+  var request: URLRequest { webSocket.request }
+  var isConnected: Bool { webSocket.isConnected }
+  var callbackQueue: DispatchQueue { webSocket.callbackQueue }
+  weak var delegate: WebSocketEngineDelegate?
+
+  init(request: URLRequest, callbackQueue: DispatchQueue) {
+    self.webSocket = Starscream.WebSocket(request: request)
+    webSocket.delegate = self
+    webSocket.callbackQueue = callbackQueue
+  }
+
+  func connect() {
+    webSocket.connect()
+  }
+
+  func disconnect() {
+    webSocket.disconnect(forceTimeout: 0)
+  }
```

Comment:
I'm not if this will be a case, but the bug I worked on reconnection on relogin we need to call delegate here with disconnect. Please check the latest version.
The same for `URLSessionWebSocketEngine`.

Reply:
`websocketDidDisconnect` is not called automatically, when you disconnect?

Reply:
I thought the same, but sometimes it's too late. Because we do force disconnect, I think it's good to change the state immediately, but probably in our new v3 connection states will be handled more carefully.

Reply:
You can keep it as is, but can you add a note there that the disconnected state can be delayed.

Reply:
This will be solved as a part of CIS-164